bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5712,10 +5733,13 @@ void ReplicaImp::sendResponses(PrePrepareMsg *ppMsg, IRequestsHandler::Execution
                                                                     currentPrimary(),
                                                                     req.outReply,
                                                                     req.outActualReplySize,
+                                                                    req.reqIndexInBatch,
                                                                     0,
                                                                     executionResult);
-    send(replyMsg.get(), req.clientId);
-    free(req.outReply);
+    if (replyMsg) {
+      send(replyMsg.get(), req.clientId);
+      free(req.outReply);
+    }
```

Comment:
If you add the 'if' condition, you need to handle the 'else' one, too. At least, add the log.

Reply:
Added logs